libs/service-portal/access-control/delegations/src/components/access/access.utils.ts

```diff
@@ -0,0 +1,108 @@
+import { AuthApiScopeGroup } from '@island.is/api/schema'
+import {
+  GROUP_PREFIX,
+  Scope,
+  SCOPE_PREFIX,
+  AuthScopeTree,
+  AUTH_API_SCOPE_GROUP_TYPE,
+  ScopeGroup,
+  AccessFormScope,
+  MappedScope,
+} from './access.types'
+
+/**
+ * Checks if scope is a scope group type
+ */
+export const isApiScopeGroup = (scope: Scope): scope is ScopeGroup =>
+  scope.__typename === AUTH_API_SCOPE_GROUP_TYPE
+
+/**
+ * Flattens out scope group into a list of scopes and extends each scope with model property for form state
+ * @param apiScopeGroup Scope group to flatten
+ * @param startIndex Index to create unique key for form
+ */
+export const flattenAndExtendApiScopeGroup = (
+  apiScopeGroup: AuthApiScopeGroup,
+  startIndex: number,
+): Scope[] => [
+  // Scope parent of the children
+  {
+    ...apiScopeGroup,
+    model: `${GROUP_PREFIX}.${startIndex}`,
+  },
+  // Scope children
+  ...(apiScopeGroup.children?.map((scope, childIndex) => ({
+    ...scope,
+    model: `${SCOPE_PREFIX}.${startIndex + childIndex}`,
+  })) || []),
```

Comment:
I would say yes and no. I tried to reuse the older code as we discussed. The AccessItem component expects data in this format and that is why I did it in this way. I remember when doing this that I thought to my self. Hmmm this is not that more simpler 😅  
I would clearly like to refactor this properly and do this a lot simpler. For now I think we should do it as a separate task, because it could take a little time which we don't have right? 
What do you think?

Reply:
Discussed offline. Decide to refactor later, see task
https://www.notion.so/Refactor-AccessItem-and-AccessForm-components-e7e276d9eca2423abb4b343be7c14c90